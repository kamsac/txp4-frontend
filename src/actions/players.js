import dotenvConfiguration from '../dotenv-configuration';
import PlayerResourceMock from '../resources/players/mock';
import PlayerResource from '../resources/players';

export const PLAYER_SUCCESS = 'PLAYER_SUCCESS';
export const PLAYER_REQUEST = 'PLAYER_REQUEST';
export const PLAYER_FAILURE = 'PLAYER_FAILURE';

export function requestPlayer() {
  return {
    type: PLAYER_REQUEST,
    payload: {
      isFetching: true,
    },
  };
}

export function receivePlayer(player) {
  return {
    type: PLAYER_SUCCESS,
    payload: {
      isFetching: false,
      player,
    },
  };
}

export function playerError() {
  return {
    type: PLAYER_FAILURE,
    payload: {
      isFetching: false,
    },
  };
}

export function loadPlayer(playerLogin) {
  return (dispatch) => {
    dispatch(requestPlayer());

    const resource = dotenvConfiguration.API_URL ? PlayerResource : PlayerResourceMock;

    resource.getPlayer(playerLogin)
      .then((response) => {
        dispatch(receivePlayer(response.data.player));
      })
      .catch(() => {
        dispatch(playerError());
      });
  };
}
