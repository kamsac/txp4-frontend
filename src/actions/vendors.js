import VendorsResource from '../resources/vendors/mock';

export const REQUEST_VENDORS = 'vendors:requestVendors';
export function requestVendors() {
  return dispatch => {
    return dispatch(fetchVendors());
  }
}

function fetchVendors() {
  return dispatch => {
    VendorsResource.getVendors((vendors) => {
      dispatch(receiveVendors(vendors));
    });
  }
}

export const RECEIVE_VENDORS = 'vendors:fetchVendorsSuccess';
export function receiveVendors(vendors) {
  return {
    type: RECEIVE_VENDORS,
    vendors: vendors,
  }
}
