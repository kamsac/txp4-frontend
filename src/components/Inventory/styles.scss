@import '../../variables';

$inventory-item-size: 100px;

.Inventory {

}

.Inventory-items {
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: repeat(auto-fill, minmax($inventory-item-size, min-content));
}

.Inventory-item {
  background-color: #ccc;
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  border: $border-width solid #000;
  border-radius: $border-radius;
  width: $inventory-item-size;
  height: $inventory-item-size;
}

.Inventory-item--engine {
  background-image: url('../../images/items/engine-item-placeholder.png');
}

.Inventory-item--transmission {
  background-image: url('../../images/items/transmission-item-placeholder.png');
}

.Inventory-item--tire {
  background-image: url('../../images/items/tire-item-placeholder.png');
}

.Inventory-item--empty {
  background-color: #888;
}
