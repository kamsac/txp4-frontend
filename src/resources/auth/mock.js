export const ACCESS_TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiaWF0IjoxNTE2MjM5MDIyLCJsb2dpbiI6Imx5bWFrIiwibmljayI6IiRzJG8kYTYwbCRjODJ5JGZhM20kYzgyYSRhNjBrIn0.aaKzXEZP2tovhBAqPJYBdmbLzKjfdBsuX3FaB58JZvE';

const AuthResourceMock = {
  login() {
    return new Promise((resolve) => {
      const response = {
        data: {
          token: ACCESS_TOKEN,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default AuthResourceMock;
