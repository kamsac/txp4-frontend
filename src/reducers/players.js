import inventoryReducer from './inventory';
import { PLAYER_FAILURE, PLAYER_REQUEST, PLAYER_SUCCESS } from '../actions/players';
import {
  INVENTORY_REQUEST,
  INVENTORY_FAILURE,
  INVENTORY_SUCCESS,
  EQUIP_REQUEST,
  EQUIP_SUCCESS,
  EQUIP_FAILURE,
} from '../actions/inventory';

const initialState = {
  isFetching: false,
  items: {},
};

export default function playersReducer(state = initialState, action) {
  switch (action.type) {
    case PLAYER_REQUEST: {
      return Object.assign({}, state, {
        isFetching: action.payload.isFetching,
      });
    }
    case PLAYER_SUCCESS: {
      const { isFetching, player } = action.payload;
      return Object.assign({}, state, {
        isFetching,
        items: Object.assign({}, state.items, {
          [player.login]: {
            login: player.login,
            nick: player.nick,
            inventory: {
              isFetching: false,
              value: player.inventory,
            },
          },
        }),
      });
    }
    case PLAYER_FAILURE: {
      const { isFetching } = action.payload;
      return Object.assign({}, state, {
        isFetching,
      });
    }

    case INVENTORY_REQUEST:
    case INVENTORY_SUCCESS:
    case INVENTORY_FAILURE:
    case EQUIP_REQUEST:
    case EQUIP_SUCCESS:
    case EQUIP_FAILURE: {
      const { playerLogin } = action.payload;

      const nextState = Object.assign({}, state);

      if (!nextState.items[playerLogin]) {
        nextState.items[playerLogin] = {};
      }

      const inventory = inventoryReducer(nextState.items[playerLogin].inventory, action);
      nextState.items[playerLogin].inventory = inventory;
      return nextState;
    }
    default:
      return state;
  }
}

export function getPlayer(state, playerLogin) {
  if (
    state.players &&
    state.players.items
  ) {
    return state.players.items[playerLogin];
  }

  return null;
}

export function getPlayerNick(state, playerLogin) {
  if (
    state.players &&
    state.players.items &&
    state.players.items[playerLogin]
  ) {
    return state.players.items[playerLogin].nick;
  }

  return null;
}
