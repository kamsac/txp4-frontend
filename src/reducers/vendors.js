import {RECEIVE_VENDORS} from '../actions/vendors';

const initialValue = [];

export function vendorsReducer(state = initialValue, action) {
  switch(action.type) {
    case RECEIVE_VENDORS:
      return [...action.vendors];
    default:
      return state;
  }
}
