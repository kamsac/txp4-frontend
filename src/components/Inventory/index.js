import React from 'react';
import './styles.scss';
import InventoryItemDetails from '../InventoryItemDetails';
import EquippedItems from '../EquippedItems';
import InventoryItem from '../InventoryItem';

export default class Inventory extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.requestItems();
    this.props.requestVendors();
  }

  render() {
    const items = this.props.items
      .filter((item) => !item.isEquipped)
      .map((item) => {
        return (
          <InventoryItem
            key={item.id}
            item={item}
            selectedItem={this.props.selectedItem}
            onSelectItem={this.props.onSelectItem}
            vendors={this.props.vendors}
          />
        );
      });

    return (
      <div className="Inventory">
        <EquippedItems
          items={this.props.items}
          equipRegions={this.props.equipRegions}
          vendors={this.props.vendors}
        />
        <div className="InventoryItems">
          {items}
        </div>
        <InventoryItemDetails
          item={this.props.selectedItem}
          onEquipItem={this.props.onEquipItem}
          vendors={this.props.vendors}
        />
      </div>
    );
  }
}
