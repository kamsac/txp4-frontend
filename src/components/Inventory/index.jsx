import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';
import InventoryItemDetails from '../InventoryItemDetails/index';
import EquippedItems from '../EquippedItems/index';
import InventoryItem from '../InventoryItem/index';
import { ItemPropTypesShape, VendorPropTypesShape } from '../../prop-types';

export default class Inventory extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedItem: null,
    };

    this.selectItem = this.selectItem.bind(this);
  }

  componentDidMount() {
    this.props.loadPlayerInventory();
    this.props.requestVendors();
  }

  selectItem(item) {
    this.setState({
      selectedItem: item,
    });
  }

  render() {
    if (!this.props.items || !this.props.vendors) {
      return <div />;
    }

    const items = this.props.items
      .filter(item => !item.isEquipped)
      .map(item => (
        <InventoryItem
          key={item.id}
          item={item}
          selectedItem={this.state.selectedItem}
          onSelectItem={this.selectItem}
          vendors={this.props.vendors}
        />
      ));

    return (
      <div className="Inventory">
        <EquippedItems
          items={this.props.items}
          equipRegions={this.props.equipRegions}
          vendors={this.props.vendors}
        />
        <div className="InventoryItems">
          {items}
        </div>
        <InventoryItemDetails
          item={this.state.selectedItem}
          onEquipItem={this.props.onEquipItem}
          vendors={this.props.vendors}
          showControls={this.props.isUserOwner}
        />
      </div>
    );
  }
}

Inventory.propTypes = {
  items: PropTypes.arrayOf(ItemPropTypesShape),
  vendors: PropTypes.arrayOf(VendorPropTypesShape),
  loadPlayerInventory: PropTypes.func.isRequired,
  requestVendors: PropTypes.func.isRequired,
  equipRegions: PropTypes.arrayOf(PropTypes.string).isRequired,
  onEquipItem: PropTypes.func.isRequired,
  isUserOwner: PropTypes.bool.isRequired,
};

Inventory.defaultProps = {
  items: null,
  vendors: null,
};
