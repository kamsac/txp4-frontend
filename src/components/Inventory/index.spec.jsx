import React from 'react';
import { shallow } from 'enzyme';
import Inventory from './index';
import { ITEMS } from '../../resources/players/mock';
import { VENDORS } from '../../resources/vendors/mock';

test('<Inventory>', () => {
  const component = shallow(<Inventory
    items={ITEMS}
    vendors={VENDORS}
    loadPlayerInventory={jest.fn()}
    loadVendors={jest.fn()}
    equipRegions={['tire', 'engine', 'transmission']}
    onEquipItem={jest.fn()}
    isUserOwner
  />);
  expect(component).toMatchSnapshot();
});
