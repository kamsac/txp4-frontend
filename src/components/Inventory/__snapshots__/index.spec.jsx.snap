// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inventory> 1`] = `
<div
  className="Inventory"
>
  <EquippedItems
    equipRegions={
      Array [
        "tire",
        "engine",
        "transmission",
      ]
    }
    items={
      Array [
        Object {
          "equipRegion": "tire",
          "id": "a1a1a1",
          "isEquipped": true,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1,
            },
            Object {
              "name": "acceleration",
              "value": 1.2,
            },
            Object {
              "name": "steering",
              "value": 1.1,
            },
            Object {
              "name": "breaking",
              "value": 1.2,
            },
            Object {
              "name": "grip",
              "value": 1.5,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Tire",
          "tier": 3,
          "vendorId": "aaa",
        },
        Object {
          "equipRegion": "tire",
          "id": "b2b2b2",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 0.4,
            },
            Object {
              "name": "acceleration",
              "value": 1,
            },
            Object {
              "name": "steering",
              "value": 1.4,
            },
            Object {
              "name": "breaking",
              "value": 2,
            },
            Object {
              "name": "grip",
              "value": 3,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Less tired",
          "tier": 1,
          "vendorId": "bbb",
        },
        Object {
          "equipRegion": "engine",
          "id": "c3c3c3",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 2,
            },
            Object {
              "name": "acceleration",
              "value": 1.8,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1,
            },
            Object {
              "name": "grip",
              "value": 0.2,
            },
            Object {
              "name": "gravity",
              "value": 1.5,
            },
          ],
          "name": "Shit engine",
          "tier": 2,
          "vendorId": "ccc",
        },
        Object {
          "equipRegion": "tire",
          "id": "d4d4d4",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1,
            },
            Object {
              "name": "acceleration",
              "value": 1,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1.1,
            },
            Object {
              "name": "grip",
              "value": 0.8,
            },
            Object {
              "name": "gravity",
              "value": 0.9,
            },
          ],
          "name": "Some tire",
          "tier": 2,
          "vendorId": "ddd",
        },
        Object {
          "equipRegion": "tire",
          "id": "e5e5e5",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1,
            },
            Object {
              "name": "acceleration",
              "value": 1.2,
            },
            Object {
              "name": "steering",
              "value": 1.1,
            },
            Object {
              "name": "breaking",
              "value": 1.2,
            },
            Object {
              "name": "grip",
              "value": 1.5,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Tire",
          "tier": 3,
          "vendorId": "aaa",
        },
        Object {
          "equipRegion": "engine",
          "id": "f6f6f6",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1.5,
            },
            Object {
              "name": "acceleration",
              "value": 3,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1,
            },
            Object {
              "name": "grip",
              "value": 1,
            },
            Object {
              "name": "gravity",
              "value": 1.5,
            },
          ],
          "name": "Duper engine",
          "tier": 2,
          "vendorId": "aaa",
        },
        Object {
          "equipRegion": "tire",
          "id": "f7f7f7",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1,
            },
            Object {
              "name": "acceleration",
              "value": 1.2,
            },
            Object {
              "name": "steering",
              "value": 1.1,
            },
            Object {
              "name": "breaking",
              "value": 1.2,
            },
            Object {
              "name": "grip",
              "value": 1.5,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Tire",
          "tier": 3,
          "vendorId": "aaa",
        },
        Object {
          "equipRegion": "tire",
          "id": "g8g8g8",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 0.4,
            },
            Object {
              "name": "acceleration",
              "value": 1,
            },
            Object {
              "name": "steering",
              "value": 1.4,
            },
            Object {
              "name": "breaking",
              "value": 2,
            },
            Object {
              "name": "grip",
              "value": 3,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Less tired",
          "tier": 1,
          "vendorId": "bbb",
        },
        Object {
          "equipRegion": "engine",
          "id": "h9h9h9",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 2,
            },
            Object {
              "name": "acceleration",
              "value": 1.8,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1,
            },
            Object {
              "name": "grip",
              "value": 0.2,
            },
            Object {
              "name": "gravity",
              "value": 1.5,
            },
          ],
          "name": "Shit engine",
          "tier": 2,
          "vendorId": "ccc",
        },
        Object {
          "equipRegion": "tire",
          "id": "i0i0i0",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1,
            },
            Object {
              "name": "acceleration",
              "value": 1,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1.1,
            },
            Object {
              "name": "grip",
              "value": 0.8,
            },
            Object {
              "name": "gravity",
              "value": 0.9,
            },
          ],
          "name": "Some tire",
          "tier": 2,
          "vendorId": "ddd",
        },
        Object {
          "equipRegion": "transmission",
          "id": "j1j1j1",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 0.7,
            },
            Object {
              "name": "acceleration",
              "value": 1.5,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1,
            },
            Object {
              "name": "grip",
              "value": 1,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Transmission",
          "tier": 3,
          "vendorId": "ccc",
        },
      ]
    }
    vendors={
      Array [
        Object {
          "color": "#8f0",
          "id": "aaa",
          "name": "Super",
        },
        Object {
          "color": "#f80",
          "id": "bbb",
          "name": "Safe",
        },
        Object {
          "color": "#800",
          "id": "ccc",
          "name": "Heavy",
        },
        Object {
          "color": "#08f",
          "id": "ddd",
          "name": "Light",
        },
      ]
    }
  />
  <div
    className="InventoryItems"
  >
    <InventoryItem
      item={
        Object {
          "equipRegion": "tire",
          "id": "b2b2b2",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 0.4,
            },
            Object {
              "name": "acceleration",
              "value": 1,
            },
            Object {
              "name": "steering",
              "value": 1.4,
            },
            Object {
              "name": "breaking",
              "value": 2,
            },
            Object {
              "name": "grip",
              "value": 3,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Less tired",
          "tier": 1,
          "vendorId": "bbb",
        }
      }
      key="b2b2b2"
      onSelectItem={[Function]}
      selectedItem={null}
      vendors={
        Array [
          Object {
            "color": "#8f0",
            "id": "aaa",
            "name": "Super",
          },
          Object {
            "color": "#f80",
            "id": "bbb",
            "name": "Safe",
          },
          Object {
            "color": "#800",
            "id": "ccc",
            "name": "Heavy",
          },
          Object {
            "color": "#08f",
            "id": "ddd",
            "name": "Light",
          },
        ]
      }
    />
    <InventoryItem
      item={
        Object {
          "equipRegion": "engine",
          "id": "c3c3c3",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 2,
            },
            Object {
              "name": "acceleration",
              "value": 1.8,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1,
            },
            Object {
              "name": "grip",
              "value": 0.2,
            },
            Object {
              "name": "gravity",
              "value": 1.5,
            },
          ],
          "name": "Shit engine",
          "tier": 2,
          "vendorId": "ccc",
        }
      }
      key="c3c3c3"
      onSelectItem={[Function]}
      selectedItem={null}
      vendors={
        Array [
          Object {
            "color": "#8f0",
            "id": "aaa",
            "name": "Super",
          },
          Object {
            "color": "#f80",
            "id": "bbb",
            "name": "Safe",
          },
          Object {
            "color": "#800",
            "id": "ccc",
            "name": "Heavy",
          },
          Object {
            "color": "#08f",
            "id": "ddd",
            "name": "Light",
          },
        ]
      }
    />
    <InventoryItem
      item={
        Object {
          "equipRegion": "tire",
          "id": "d4d4d4",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1,
            },
            Object {
              "name": "acceleration",
              "value": 1,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1.1,
            },
            Object {
              "name": "grip",
              "value": 0.8,
            },
            Object {
              "name": "gravity",
              "value": 0.9,
            },
          ],
          "name": "Some tire",
          "tier": 2,
          "vendorId": "ddd",
        }
      }
      key="d4d4d4"
      onSelectItem={[Function]}
      selectedItem={null}
      vendors={
        Array [
          Object {
            "color": "#8f0",
            "id": "aaa",
            "name": "Super",
          },
          Object {
            "color": "#f80",
            "id": "bbb",
            "name": "Safe",
          },
          Object {
            "color": "#800",
            "id": "ccc",
            "name": "Heavy",
          },
          Object {
            "color": "#08f",
            "id": "ddd",
            "name": "Light",
          },
        ]
      }
    />
    <InventoryItem
      item={
        Object {
          "equipRegion": "tire",
          "id": "e5e5e5",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1,
            },
            Object {
              "name": "acceleration",
              "value": 1.2,
            },
            Object {
              "name": "steering",
              "value": 1.1,
            },
            Object {
              "name": "breaking",
              "value": 1.2,
            },
            Object {
              "name": "grip",
              "value": 1.5,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Tire",
          "tier": 3,
          "vendorId": "aaa",
        }
      }
      key="e5e5e5"
      onSelectItem={[Function]}
      selectedItem={null}
      vendors={
        Array [
          Object {
            "color": "#8f0",
            "id": "aaa",
            "name": "Super",
          },
          Object {
            "color": "#f80",
            "id": "bbb",
            "name": "Safe",
          },
          Object {
            "color": "#800",
            "id": "ccc",
            "name": "Heavy",
          },
          Object {
            "color": "#08f",
            "id": "ddd",
            "name": "Light",
          },
        ]
      }
    />
    <InventoryItem
      item={
        Object {
          "equipRegion": "engine",
          "id": "f6f6f6",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1.5,
            },
            Object {
              "name": "acceleration",
              "value": 3,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1,
            },
            Object {
              "name": "grip",
              "value": 1,
            },
            Object {
              "name": "gravity",
              "value": 1.5,
            },
          ],
          "name": "Duper engine",
          "tier": 2,
          "vendorId": "aaa",
        }
      }
      key="f6f6f6"
      onSelectItem={[Function]}
      selectedItem={null}
      vendors={
        Array [
          Object {
            "color": "#8f0",
            "id": "aaa",
            "name": "Super",
          },
          Object {
            "color": "#f80",
            "id": "bbb",
            "name": "Safe",
          },
          Object {
            "color": "#800",
            "id": "ccc",
            "name": "Heavy",
          },
          Object {
            "color": "#08f",
            "id": "ddd",
            "name": "Light",
          },
        ]
      }
    />
    <InventoryItem
      item={
        Object {
          "equipRegion": "tire",
          "id": "f7f7f7",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1,
            },
            Object {
              "name": "acceleration",
              "value": 1.2,
            },
            Object {
              "name": "steering",
              "value": 1.1,
            },
            Object {
              "name": "breaking",
              "value": 1.2,
            },
            Object {
              "name": "grip",
              "value": 1.5,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Tire",
          "tier": 3,
          "vendorId": "aaa",
        }
      }
      key="f7f7f7"
      onSelectItem={[Function]}
      selectedItem={null}
      vendors={
        Array [
          Object {
            "color": "#8f0",
            "id": "aaa",
            "name": "Super",
          },
          Object {
            "color": "#f80",
            "id": "bbb",
            "name": "Safe",
          },
          Object {
            "color": "#800",
            "id": "ccc",
            "name": "Heavy",
          },
          Object {
            "color": "#08f",
            "id": "ddd",
            "name": "Light",
          },
        ]
      }
    />
    <InventoryItem
      item={
        Object {
          "equipRegion": "tire",
          "id": "g8g8g8",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 0.4,
            },
            Object {
              "name": "acceleration",
              "value": 1,
            },
            Object {
              "name": "steering",
              "value": 1.4,
            },
            Object {
              "name": "breaking",
              "value": 2,
            },
            Object {
              "name": "grip",
              "value": 3,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Less tired",
          "tier": 1,
          "vendorId": "bbb",
        }
      }
      key="g8g8g8"
      onSelectItem={[Function]}
      selectedItem={null}
      vendors={
        Array [
          Object {
            "color": "#8f0",
            "id": "aaa",
            "name": "Super",
          },
          Object {
            "color": "#f80",
            "id": "bbb",
            "name": "Safe",
          },
          Object {
            "color": "#800",
            "id": "ccc",
            "name": "Heavy",
          },
          Object {
            "color": "#08f",
            "id": "ddd",
            "name": "Light",
          },
        ]
      }
    />
    <InventoryItem
      item={
        Object {
          "equipRegion": "engine",
          "id": "h9h9h9",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 2,
            },
            Object {
              "name": "acceleration",
              "value": 1.8,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1,
            },
            Object {
              "name": "grip",
              "value": 0.2,
            },
            Object {
              "name": "gravity",
              "value": 1.5,
            },
          ],
          "name": "Shit engine",
          "tier": 2,
          "vendorId": "ccc",
        }
      }
      key="h9h9h9"
      onSelectItem={[Function]}
      selectedItem={null}
      vendors={
        Array [
          Object {
            "color": "#8f0",
            "id": "aaa",
            "name": "Super",
          },
          Object {
            "color": "#f80",
            "id": "bbb",
            "name": "Safe",
          },
          Object {
            "color": "#800",
            "id": "ccc",
            "name": "Heavy",
          },
          Object {
            "color": "#08f",
            "id": "ddd",
            "name": "Light",
          },
        ]
      }
    />
    <InventoryItem
      item={
        Object {
          "equipRegion": "tire",
          "id": "i0i0i0",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 1,
            },
            Object {
              "name": "acceleration",
              "value": 1,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1.1,
            },
            Object {
              "name": "grip",
              "value": 0.8,
            },
            Object {
              "name": "gravity",
              "value": 0.9,
            },
          ],
          "name": "Some tire",
          "tier": 2,
          "vendorId": "ddd",
        }
      }
      key="i0i0i0"
      onSelectItem={[Function]}
      selectedItem={null}
      vendors={
        Array [
          Object {
            "color": "#8f0",
            "id": "aaa",
            "name": "Super",
          },
          Object {
            "color": "#f80",
            "id": "bbb",
            "name": "Safe",
          },
          Object {
            "color": "#800",
            "id": "ccc",
            "name": "Heavy",
          },
          Object {
            "color": "#08f",
            "id": "ddd",
            "name": "Light",
          },
        ]
      }
    />
    <InventoryItem
      item={
        Object {
          "equipRegion": "transmission",
          "id": "j1j1j1",
          "isEquipped": false,
          "modifiers": Array [
            Object {
              "name": "maxSpeed",
              "value": 0.7,
            },
            Object {
              "name": "acceleration",
              "value": 1.5,
            },
            Object {
              "name": "steering",
              "value": 1,
            },
            Object {
              "name": "breaking",
              "value": 1,
            },
            Object {
              "name": "grip",
              "value": 1,
            },
            Object {
              "name": "gravity",
              "value": 1,
            },
          ],
          "name": "Transmission",
          "tier": 3,
          "vendorId": "ccc",
        }
      }
      key="j1j1j1"
      onSelectItem={[Function]}
      selectedItem={null}
      vendors={
        Array [
          Object {
            "color": "#8f0",
            "id": "aaa",
            "name": "Super",
          },
          Object {
            "color": "#f80",
            "id": "bbb",
            "name": "Safe",
          },
          Object {
            "color": "#800",
            "id": "ccc",
            "name": "Heavy",
          },
          Object {
            "color": "#08f",
            "id": "ddd",
            "name": "Light",
          },
        ]
      }
    />
  </div>
  <InventoryItemDetails
    item={null}
    onEquipItem={[MockFunction]}
    showControls={true}
    vendors={
      Array [
        Object {
          "color": "#8f0",
          "id": "aaa",
          "name": "Super",
        },
        Object {
          "color": "#f80",
          "id": "bbb",
          "name": "Safe",
        },
        Object {
          "color": "#800",
          "id": "ccc",
          "name": "Heavy",
        },
        Object {
          "color": "#08f",
          "id": "ddd",
          "name": "Light",
        },
      ]
    }
  />
</div>
`;
